### PR TITLE
Support 'petersburg' in genesis files (#800)

### LINKED ISSUE #800: Support 'petersburg' in genesis files
The fork between Constantinople and Istanbul is known by two different names.  The Reference Tests refer to it as ConstantinopleFix, and hive integration tests refer to it as Petersburg.  We should allow genesis files to support both.

* Either `constantinopleFixBlock` or `petersburgBlock` can be specified in a genesis file
* It is an error to specify both
* default .json files should migrate to `petersburgBlock`
* it is not necessary to change all code occurrences of `constantinopleFix`. If done it should be in a separate single-purpose PR.

### PR DESCRIPTION
**Pull request by br0tchain**
## PR description
Support 'petersburg' in genesis files (#800)
feat: change `constantinopleFixBlock` to `petersburgBlock` in genesis configuration files

## Fixed Issue(s)

fixes #800 

Signed-off-by: Alexandre PARIS-VERGNE <alexpv14@gmail.com>

### CODE REVIEW

**Comment by shemnon** [comment]
We need to be able to support either `constantinopleFix` or `petersburg` in the genesis files, a simple cutover will break things.

We will also need unit tests with 4 mini json files to make sure we get what we expect: one with `petersburg` that sets the petersburg block, one with `constantinoplefix` that will set it as the petersburg block, one with both that will result in an error, and one with neither that will result in petersburg block not being set.  Probably in `org.hyperledger.besu.config.GenesisConfigFileTest`

**Comment by br0tchain** [comment]
Hi @shemnon , thanks for the review. I've addressed the issues you've raised and pushed the change. Could you please review it again ?

File: config/src/test/java/org/hyperledger/besu/config/GenesisConfigFileTest.java [hunk 1]
```diff
@@ -298,6 +298,20 @@ public void testNoOverride() {
     assertThat(config.getConfigOptions().getEvmStackSize()).isNotPresent();
   }
 
+  @Test(expected = RuntimeException.class)
```
Comment (shemnon) [inline config/src/test/java/org/hyperledger/besu/config/GenesisConfigFileTest.java:301]: We are transitioning away from `expected` in annotation to `assertThatExceptionOfType(...).isThrownBy(...).withMessage(...)`.  It reads better and makes it more clear where the exception will occur. Can you migrate to this format?

File: config/src/test/java/org/hyperledger/besu/config/GenesisConfigOptionsTest.java [hunk 2]
```diff
@@ -129,6 +130,21 @@ public void shouldGetConstantinopleFixBlockNumber() {
     assertThat(config.getConstantinopleFixBlockNumber()).hasValue(1000);
   }
 
+  @Test
+  public void shouldGetPetersburgBlockNumber() {
+    final GenesisConfigOptions config = fromConfigOptions(singletonMap("petersburgBlock", 1000));
+    assertThat(config.getConstantinopleFixBlockNumber()).hasValue(1000);
+  }
+
+  @Test(expected = RuntimeException.class)
```
Comment (shemnon) [inline config/src/test/java/org/hyperledger/besu/config/GenesisConfigOptionsTest.java:139]: Same exception pattern change request as above.

File: config/src/test/java/org/hyperledger/besu/config/GenesisConfigFileTest.java (hunk 1 above)
Comment (shemnon) [reply config/src/test/java/org/hyperledger/besu/config/GenesisConfigFileTest.java:301]: as seen here - https://github.com/hyperledger/besu/blob/98ec5777dd61aa006727611644329b0195a939cd/enclave/src/test/java/org/hyperledger/besu/enclave/EnclaveFactoryTest.java#L44

**Comment by br0tchain** [comment]
Hi @shemnon , I have fixed the style issues as you requested. Could you please confirm everything is fine ? Thanks

**Pull request merged**